spec/pushmi_pullyu/cli_spec.rb

```diff
@@ -0,0 +1,218 @@
+require 'spec_helper'
+require 'tempfile'
+
+RSpec.describe PushmiPullyu::CLI do
+  let(:cli) { described_class.instance }
+
+  describe '#run' do
+    before do
+      allow(cli).to receive(:start_server)
+      allow(cli).to receive(:start_server_as_daemon)
+    end
+
+    it 'starts working loop' do
+      cli.run
+
+      expect(cli).to have_received(:start_server).once
+      expect(cli).not_to have_received(:start_server_as_daemon)
+    end
+
+    context 'should run as daemon' do
+      it 'starts working loop as daemon' do
+        PushmiPullyu.options[:daemonize] = true
+
+        cli.run
+
+        expect(cli).to have_received(:start_server_as_daemon).once
+        expect(cli).not_to have_received(:start_server)
+      end
+    end
+  end
+
+  describe '#start_server' do
+    it 'starts run tick loop' do
+      allow(cli).to receive(:setup_signal_traps)
+      allow(cli).to receive(:setup_log)
+      allow(cli).to receive(:print_banner)
+      allow(cli).to receive(:run_tick_loop)
+
+      cli.start_server
+
+      expect(cli).to have_received(:setup_signal_traps).once
+      expect(cli).to have_received(:setup_log).once
+      expect(cli).to have_received(:print_banner).once
+      expect(cli).to have_received(:run_tick_loop).once
+    end
+  end
+
+  describe '#parse' do
+    let!(:opts) { PushmiPullyu.options.dup }
+
+    after do
+      PushmiPullyu.options = opts
+    end
```

Comment:
This is what looks fishy to me. We're copying data out of an object and then writing it back in afterwards in a way that is really not demeter-friendly. We should encapsulate this interaction:

`PushmiPullyu.reset_options`